File not found exception on dependency quarkus-resteasy-reactive-deployment-1.11.0.Final.jar

**Describe the bug**
Cant start the application due to quarkus cant found a file (the file exists, i cant figure out what is happening)

**Expected behavior**
Application start
(ps), when i ran the tests, the application starts,strange

**Actual behavior**
(Describe the actual behavior clearly and concisely.)
```
[INFO] Nothing to compile - all classes are up to date
Listening for transport dt_socket at address: 5005
2021-01-21 19:34:56,998 INFO  [org.jbo.threads] (main) JBoss Threads version 3.2.0.Final
2021-01-21 19:34:57,181 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.resteasy.reactive.server.deployment.devconsole.DevConsoleProcessor#collectScores threw an exception: java.nio.file.FileSystemNotFoundException: C:\Users\Pedro%20Esnaola\.m2\repository\io\quarkus\quarkus-resteasy-reactive-deployment\1.11.0.Final\quarkus-resteasy-reactive-deployment-1.11.0.Final.jar
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:118)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:136)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:406)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:47)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:32)
	at io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem.<init>(DevConsoleRuntimeTemplateInfoBuildItem.java:39)
	at io.quarkus.resteasy.reactive.server.deployment.devconsole.DevConsoleProcessor.collectScores(DevConsoleProcessor.java:12)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	[error]: Build step io.quarkus.arc.deployment.devconsole.DevConsoleProcessor#collectBeanInfo threw an exception: java.nio.file.FileSystemNotFoundException: C:\Users\Pedro%20Esnaola\.m2\repository\io\quarkus\quarkus-arc-deployment\1.11.0.Final\quarkus-arc-deployment-1.11.0.Final.jar
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:118)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:136)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:406)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:47)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:32)
	at io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem.<init>(DevConsoleRuntimeTemplateInfoBuildItem.java:39)
	at io.quarkus.arc.deployment.devconsole.DevConsoleProcessor.collectBeanInfo(DevConsoleProcessor.java:33)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:274)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:150)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:57)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:78)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:372)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:128)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:81)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.resteasy.reactive.server.deployment.devconsole.DevConsoleProcessor#collectScores threw an exception: java.nio.file.FileSystemNotFoundException: C:\Users\Pedro%20Esnaola\.m2\repository\io\quarkus\quarkus-resteasy-reactive-deployment\1.11.0.Final\quarkus-resteasy-reactive-deployment-1.11.0.Final.jar
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:118)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:136)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:406)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:47)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:32)
	at io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem.<init>(DevConsoleRuntimeTemplateInfoBuildItem.java:39)
	at io.quarkus.resteasy.reactive.server.deployment.devconsole.DevConsoleProcessor.collectScores(DevConsoleProcessor.java:12)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	[error]: Build step io.quarkus.arc.deployment.devconsole.DevConsoleProcessor#collectBeanInfo threw an exception: java.nio.file.FileSystemNotFoundException: C:\Users\Pedro%20Esnaola\.m2\repository\io\quarkus\quarkus-arc-deployment\1.11.0.Final\quarkus-arc-deployment-1.11.0.Final.jar
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:118)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:136)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:406)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:47)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:32)
	at io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem.<init>(DevConsoleRuntimeTemplateInfoBuildItem.java:39)
	at io.quarkus.arc.deployment.devconsole.DevConsoleProcessor.collectBeanInfo(DevConsoleProcessor.java:33)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:148)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:272)
	... 9 more
	Suppressed: java.nio.file.FileSystemNotFoundException: C:\Users\Pedro%20Esnaola\.m2\repository\io\quarkus\quarkus-arc-deployment\1.11.0.Final\quarkus-arc-deployment-1.11.0.Final.jar
		at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:118)
		at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:136)
		at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:406)
		at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:47)
		at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:32)
		at io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem.<init>(DevConsoleRuntimeTemplateInfoBuildItem.java:39)
		at io.quarkus.arc.deployment.devconsole.DevConsoleProcessor.collectBeanInfo(DevConsoleProcessor.java:33)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
		at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
		at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
		at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
		at java.base/java.lang.Thread.run(Thread.java:829)
		at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.nio.file.FileSystemNotFoundException: C:\Users\Pedro%20Esnaola\.m2\repository\io\quarkus\quarkus-resteasy-reactive-deployment\1.11.0.Final\quarkus-resteasy-reactive-deployment-1.11.0.Final.jar
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:118)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:136)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:406)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:47)
	at io.quarkus.deployment.util.ArtifactInfoUtil.groupIdAndArtifactId(ArtifactInfoUtil.java:32)
	at io.quarkus.devconsole.spi.DevConsoleRuntimeTemplateInfoBuildItem.<init>(DevConsoleRuntimeTemplateInfoBuildItem.java:39)
	at io.quarkus.resteasy.reactive.server.deployment.devconsole.DevConsoleProcessor.collectScores(DevConsoleProcessor.java:12)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

2021-01-21 19:34:57,181 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
```


**To Reproduce**

I just build a project using quarkus plugin for intellij

Or attach an archive containing the reproducer to the issue.

Steps to reproduce the behavior:
1. 
2. 
3. 

**Configuration**
```properties
# Add your application.properties here, if applicable.
none
```



**Screenshots**
(If applicable, add screenshots to help explain your problem.)


![image](https://user-images.githubusercontent.com/49621948/105421232-8d0bc600-5c20-11eb-84f4-4a5c9939ab41.png)
(the file exists)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 

**Additional context**
(Add any other context about the problem here.)


resteasy 
resteasy-reactive

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/DevConsoleProcessor.java
@@ -6,6 +6,7 @@ import java.io.Reader;
 import java.io.StringReader;
 import java.net.URL;
 import java.net.URLConnection;
+import java.net.URLDecoder;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.FileSystem;
 import java.nio.file.FileSystems;
@@ -432,7 +433,7 @@ public class DevConsoleProcessor {
                         jarPath = jarPath.substring(1).replace('/', '\\');
                     }
                     try (FileSystem fs = FileSystems
-                            .newFileSystem(Paths.get(jarPath), classLoader)) {
+                            .newFileSystem(Paths.get(URLDecoder.decode(jarPath, StandardCharsets.UTF_8.name())), classLoader)) {
                         scanTemplates(fs, devTemplatePaths);
                     }
                 }

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/JarClassPathElement.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/JarClassPathElement.java
@@ -10,7 +10,6 @@ import java.io.InterruptedIOException;
 import java.io.UncheckedIOException;
 import java.lang.reflect.Method;
 import java.net.MalformedURLException;
-import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
 import java.nio.file.Path;
@@ -210,8 +209,7 @@ public class JarClassPathElement implements ClassPathElement {
     public ProtectionDomain getProtectionDomain(ClassLoader classLoader) {
         final URL url;
         try {
-            URI uri = new URI("file", null, jarPath.getPath(), null);
-            url = uri.toURL();
+            url = jarPath.toURI().toURL();
         } catch (URISyntaxException | MalformedURLException e) {
             throw new RuntimeException("Unable to create protection domain for " + jarPath, e);
         }

